django/db/migrations/operations/base.py

```diff
@@ -98,17 +98,15 @@ def references_field(self, model_name, name, app_label=None):
         """
         return self.references_model(model_name, app_label)
 
-    def allowed_to_migrate(self, connection_alias, model):
+    def allowed_to_migrate(self, connection_alias, model, hints=None):
         """
-        Returns if we're allowed to migrate the model. Checks the router,
-        if it's a proxy, if it's managed, and if it's swapped out.
+        Returns if we're allowed to migrate the model.
         """
-        return (
-            not model._meta.proxy and
-            not model._meta.swapped and
-            model._meta.managed and
-            router.allow_migrate(connection_alias, model)
-        )
+        # Always skip if proxy, unmanaged or swapped out.
+        if model and (model._meta.proxy or model._meta.swapped or not model._meta.managed):
+            return False
+
```

Comment:
This snippet was used to prevent a backwards incompatibility so that if you never use `hints` you don't have to worry about `model == None`. After discussing this with @akaariai on IRC we concluded that always going to the router was worth the backwards incompatibility since it allows you to easily switch off migrations for any given db.

Also if a third party app ends up using ``hints`` you'll need to handle `model == None` anyway.